src/reconcile/rec_write.c

```diff
@@ -1811,24 +1813,22 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
     if (F_ISSET(r, WT_REC_IN_MEMORY))
         goto copy_image;
 
-    /* If we need to restore the page to memory, copy the disk image. */
-    if (r->cache_write_restore && multi->supd != NULL) {
-        WT_ASSERT(session, F_ISSET(r, WT_REC_EVICT) && r->leave_dirty);
-        goto copy_image;
-    }
+    if (F_ISSET(r, WT_REC_EVICT) && multi->supd != NULL) {
```

Comment:
Why only for the eviction scenario? Can you explain it in the comment? Before durable history, the checkpoint flow also can enter in that code.

Reply:
No, it can't. Before durable history. We nave save updates for checkpoint.